### PR TITLE
Set LONG_TESTS when we're in the weekly travis cron

### LINKED ISSUE #310: Schedule for long running tests in travis
Make a schedule to launch long-running tests during the weekends so we know if something is failing.

### PR DESCRIPTION
**Pull request by m09**
Fixes #310.

### CODE REVIEW

**Comment by zurk** [comment]
@m09 do you find a workaround to run them each week?

**Comment by m09** [comment]
yes it seems they keep on being triggered at the original time the cron was created, so I'll set it up today to have it be run each saturday

File: lookout/style/format/tests/test_analyzer_integration.py [hunk 1]
```diff
@@ -94,7 +94,8 @@ def tearDown(self, fs_cleanup=True):
         self.analyzer.__exit__()
 
 
-@unittest.skipUnless(os.getenv("LONG_TESTS", False),
+@unittest.skipUnless(os.getenv("LONG_TESTS", False)
+                     or os.getenv("TRAVIS_EVENT_TYPE", "other") == "cron",
```
[the remark this answers is not in the document]
Comment (vmarkovtsev) [reply lookout/style/format/tests/test_analyzer_integration.py]: We should never detect Travis in the tests. LONG_TESTS is ok.

[the remark this answers is not in the document]
Comment (m09) [reply lookout/style/format/tests/test_analyzer_integration.py]: we cannot set different env variables for normal builds and crons though.

Comment (vmarkovtsev) [inline lookout/style/format/tests/test_analyzer_integration.py]: We should not detect Travis or any other CI system.

[the remark this answers is not in the document]
Comment (zurk) [reply lookout/style/format/tests/test_analyzer_integration.py]: I have one idea, can we check day of the week inside Python and just add it to condition?

[the remark this answers is not in the document]
Comment (m09) [reply lookout/style/format/tests/test_analyzer_integration.py]: and then when we work on saturday like today travis takes 3h per run? Not workable I think

[the remark this answers is not in the document]
Comment (vmarkovtsev) [reply lookout/style/format/tests/test_analyzer_integration.py]: This is even worse :)

[the remark this answers is not in the document]
Comment (m09) [reply lookout/style/format/tests/test_analyzer_integration.py]: alternative is to have the cron on another server (ours) and use travis API to launch build with given env (if it's possible, didn't check) but I do think this alternative is x15 worse than checking for travis in tests.

**Comment by vmarkovtsev** [comment]
Hint: you can change `.travis.yml` any way you want.

File: lookout/style/format/tests/test_analyzer_integration.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zurk) [reply lookout/style/format/tests/test_analyzer_integration.py]: ok, can we check the weekday in Travis and set `LONG_TESTS=True` if it is Sunday? and then trigger Travis by cron?

[the remark this answers is not in the document]
Comment (vmarkovtsev) [reply lookout/style/format/tests/test_analyzer_integration.py]: @m09 If you can check an env in Python, then you can check the same in your Travis script instead.

[the remark this answers is not in the document]
Comment (m09) [reply lookout/style/format/tests/test_analyzer_integration.py]: yup I'll do that

[the remark this answers is not in the document]
Comment (vmarkovtsev) [reply lookout/style/format/tests/test_analyzer_integration.py]: @zurk If Travis is triggered by cron, it defines a special env var. No need for any hacks, people, this is CI 101

**Comment by m09** [comment]
I tried a few things but none of them actually triggered long tests on a cron build (to me the new conditions -`type = cron`- don't seem to work but I probably just don't speak travis). Since there is no longer a relevant change in the PR I'll close this one and we'll create a new one when we'll want to come back to the issue.

**Pull request closed without merging**